examples/next/pages/ui/hooks/useDataStore/filter/CollectionAndRecord.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { User, UserPreference } from '../models';
+import {
+  createDataStorePredicate,
+  getOverrideProps,
+  useDataStoreBinding,
+} from '@aws-amplify/ui-react/internal';
+import { Button, Collection, Flex } from '@aws-amplify/ui-react';
+import { colorFilterPredicate, userFilterPredicate } from '../utils';
+
+const CollectionAndRecord = (props) => {
+  const {
+    backgroundColor,
+    buttonColor: buttonColorProp,
+    items,
+    overrideItems,
+    overrides,
+    ...rest
+  } = props;
+
+  const userFilter = createDataStorePredicate(userFilterPredicate);
+  const usersDataStore = useDataStoreBinding({
+    criteria: userFilter,
+    model: User,
+    type: 'collection',
+  }).items;
+  const users = items !== undefined ? items : usersDataStore;
```

Comment:
We can do 
```suggestion
  const users = items ?? usersDataStore;
```

but open to opinions